src/java/org/apache/cassandra/db/commitlog/CommitLogSegmentReader.java

```diff
@@ -50,8 +50,10 @@
  */
 public class CommitLogSegmentReader implements Iterable<CommitLogSegmentReader.SyncSegment>
 {
-    public static final String ALLOW_IGNORE_SYNC_CRC = Config.PROPERTY_PREFIX + "commitlog.allow_ignore_sync_crc";
-    private static volatile boolean allowSkipSyncMarkerCrc = Boolean.getBoolean(ALLOW_IGNORE_SYNC_CRC);
+    /** @deprecated Use {@link org.apache.cassandra.config.CassandraRelevantProperties#COMMITLOG_ALLOW_IGNORE_SYNC_CRC}. */
```

Comment:
I think this deprecated comment + annotation should not be here?

Reply:
My thoughts here are - the `ALLOW_IGNORE_SYNC_CRC` is a public constant inside a public class, so it seems to me it may be related to the public API change which I would like to avoid in this patch. 
Some users may override/use the `CommitLogSegmentReader` class and its constants in their production or test code thus if we remove this constant from this class we can introduce a compilation failure once these changes get released.

Do you think it is safe to remove it?

Reply:
ok, do nothing.

Reply:
@jacek-lewandowski if we follow our deprecation rules (deprecate in one major, wait one major and remove it in the next), we should, technically, still have this one in 5.0 however weird it looks.

Reply:
But it applies to an API - I don't think this is an API. On the other hand, we have some property names defined in interfaces like the one for `Guardrails` or `SchemaUpdateHandlerProvider` which are APIs.

We can also deprecate it in 4.1.x, just like with Hadoop, which was much more significant deprecation and we decided to do that.

Reply:
discussed with @michaelsembwever on Slack, safe to remove this field (there was a misunderstanding and Mick meant that the property itself is part of the API)

Reply:
yes my bad, thanks for catching me @jacek-lewandowski .

Reply:
Removed.